nodes/src/nodes/llm_vision_openai/openai_vision.py

```
@@ -0,0 +1,209 @@
+                            model=self._model,
+                            messages=messages,
+                        )
+                    except Exception as e:
+                        exc[0] = e
+
+                t = threading.Thread(target=_invoke, daemon=True)
+                t.start()
+                t.join(timeout=hard_timeout)
+                if t.is_alive():
+                    warning(f'OpenAI Vision: inference timed out after {hard_timeout}s (attempt {attempt + 1}/{max_retries + 1}) — daemon thread still running')
+                    raise TimeoutError(f'Vision inference timed out after {hard_timeout}s (attempt {attempt + 1})')
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)